text/0000-light-dom.md

```diff
@@ -0,0 +1,293 @@
+
+##### Lazy slots
+
+The main reason for this difference is that synthetic shadow slots and light DOM slots are both _lazy_. For example:
+
+```html
+<!-- x/slottable.html -->
+<template lwc:no-shadow>
+  <template if:true={foo}>
+    <slot lwc:no-shadow></slot>
+  </template>
+</template>
+```
+
+```js
+// slottable.js
+import { LightningElement, track } from 'lwc'
+export default class Slottable extends LightningElement {
+  foo = false
+}
+```
+
+```html
+<!-- app.html -->
+<template>
+  <x-slottable>
+    <x-foo></x-foo>
+  </x-slottable>
+</template>
+```
+
+In this case, `<x-foo>` will never render at all, because it's within a `<template if:true>` block where the condition is false. Similarly, the slot contents would not render if there was no receiving slot for a given slot name, or if the target component contained no slots at all.
+
+These are all differences with native shadow slots, which render eagerly. However, these differences align with the current implementation of synthetic shadow slots. This behavior is maintained in light DOM slots for simplicity's sake in the LWC engine, as well as for the performance benefit of lazy slots.
+
+##### Events and other slot features
+
+The `slotchange` event and `::slotted` CSS pseudo-selector are both unsupported for light DOM slots.
+
+In the case of `slotchange`, it's not clear what `event.target` would be. For `::slotted`, it's not really relevant or useful since light DOM slots can be styled the same as any other light DOM content.
+
+#### Styles
+
+Until now styles used in LWC components were scoped to the component thanks to shadow DOM (or synthetic shadow DOM) style scoping. In the light DOM, component styles naturally leak out of the component; LWC doesn't do any style scoping out of the box. Developers are in charge of writing selectors that are specific enough to target the intended element or pseudo-element.
+
+To support the cases where a shadow DOM element composes a light element, light DOM styles are required to be injected to the closest root node. For a given light DOM element, if all the ancestor components are also light DOM components, the component style sheet will be injected in the document `<head>`. Otherwise, if any of the ancestors is a shadow DOM component, the style has to be injected into the closest shadow root. Upon insertion of a light DOM element, LWC does the following steps:
+
+- look up for the closest root node (`Node.prototype.getRootNode()`)
+- insert the stylesheet if not already present:
+  - if the root node is the HTML document, the style sheet is inserted as a `<style>` element in the `<head>`.
+  - if the root node is a shadow root, the stylesheet is inserted as a `<style>` element as the first shadow root child.
+
+It is important to notice that the order in which light DOM components are rendered impacts the order in which stylesheets are injected into the root node and directly influences CSS rule specificity.
```

Comment:
This part of the doc needs to be updated with more detail about how this will work for synthetic shadow, and to specify that `:host`/`:host-context()` will not be supported for light DOM.